js/components/simulation.tsx

```diff
@@ -89,6 +92,10 @@ export default class Simulation extends BaseComponent<IBaseProps, IState> {
         <div className="cross-section-container">
           <CrossSection />
         </div>
+        {
+          isMeasuringTempPressure &&
+          <TempPressureOverlay simulationStore={this.simulationStore} />
```

Comment:
If you need to pass `simulationStore`, I guess you might want to use:
```javascript
@inject("simulationStore")
@observer
```
in `TempPressureOverlay`. It's only one level between the parent and child component, so it doesn't change that much. But it's gonna be more consistent with the rest of the app.